### PR TITLE
changing routes for IceRaids and adding Check Point Files for back end and front end.

### PR DESCRIPTION
**Pull request by smcueto**
Axios call for Check Point is throwing error 404 msg.

### CODE REVIEW

File: server/models/CheckPointmodel.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+var CheckPointCollectionModel = new Schema({
```
Comment (fluxsauce) [inline server/models/CheckPointmodel.js]: `CheckPointModel`

File: server/models/CheckPointmodel.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
```
Comment (fluxsauce) [inline server/models/CheckPointmodel.js]: Use `const` if it never changes.

File: server/models/IceRaidModel.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+var mongoose = require('mongoose');
```
Comment (fluxsauce) [inline server/models/IceRaidModel.js]: Use `const` if it never changes.

File: server/models/IceRaidModel.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+var Schema = mongoose.Schema;
+
+// Define collection and schema for Items
+var iceRaidCollectionModel = new Schema({
```
Comment (fluxsauce) [inline server/models/IceRaidModel.js]: `IceRaidModel`

File: server/models/CheckPointmodel.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  checkPointZipcode: {
+      type: Number
+  },
+  checkPointCity: {
+    type: String
+  },
+  checkPointState: {
+    type: String
+  },
+  checkPointDate: {
```
Comment (fluxsauce) [inline server/models/CheckPointmodel.js]: You've got two fields for the date, date and time, and neither of them use the native `Date` storage. I'd recommend using just one field and https://docs.mongodb.com/manual/reference/bson-types/#date

File: server/models/IceRaidModel.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  iceRaidZipcode: {
+      type: Number
+  },
+  iceRaidCity: {
+    type: String
+  },
+  iceRaidState: {
+    type: String
+},
+  iceRaidDate: {
```
Comment (fluxsauce) [inline server/models/IceRaidModel.js]: You've got two fields for the date, date and time, and neither of them use the native `Date` storage. I'd recommend using just one field and https://docs.mongodb.com/manual/reference/bson-types/#date

File: server/routes/CheckServerRoutes.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    })
+    .catch(err => {
+    res.status(400).send("unable to save to database");
+    });
+});
+
+ServerPortRouter.route('/').get(function (req, res) {
+    checkPointCollectionDirectory.find(function (err, serverports){
+    if(err){
+      console.log(err);
```
Comment (fluxsauce) [inline server/routes/CheckServerRoutes.js]: Return an error rather than log

File: server/routes/CheckServerRoutes.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+const app = express();
+const ServerPortRouter = express.Router();
+const checkPointCollectionDirectory = require('../models/CheckPointModel');
+
+ServerPortRouter.route('/add').post(function (req, res) {
+  
+  const checkPointBodyRequest = new checkPointCollectionDirectory(req.body);
+  checkPointBodyRequest.save()
+    .then(checkPointBodyRequest => {
+        res.json('Server added successfully');
```
Comment (fluxsauce) [inline server/routes/CheckServerRoutes.js]: Server?

File: server/routes/CheckServerRoutes.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    })
+    .catch(err => {
+    res.status(400).send("unable to save to database");
```
Comment (fluxsauce) [inline server/routes/CheckServerRoutes.js]: Should this be a 500?

File: server/routes/IceServerRoutes.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
+const express = require('express');
+const app = express();
+const ServerPortRouter = express.Router();
+const iceCollectionDirectory = require('../models/IceRaidModel');
```
Comment (fluxsauce) [inline server/routes/IceServerRoutes.js]: Best practice is to be consistent with the naming, there's no value in calling it something different and if anything it's more confusing. `IceRaidModel` is fine.

File: server/routes/IceServerRoutes.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+      raidBodyRequest.iceRaidCity = req.body.iceRaidCity;
+      raidBodyRequest.iceRaidState = req.body.iceRaidState;
+      raidBodyRequest.iceRaidDate = req.body.iceRaidDate;
+      raidBodyRequest.iceRaidTime = req.body.iceRaidTime;
+      raidBodyRequest.save().then(raidBodyRequest => {
+          res.json(raidBodyRequest);
+    
+      })
+      .catch(err => {
+            res.status(400).send("unable to update the database");
```
Comment (fluxsauce) [inline server/routes/IceServerRoutes.js]: 500?

**Review by fluxsauce — CHANGES_REQUESTED** [review]
Good start! Here's a first pass...

**Comment by smcueto** [comment]
@fluxsauce applied 👍

**Pull request merged**